dotnet/src/Microsoft.Agents.AI.Hosting.OpenAI/Responses/Common/Id/DefaultIdGenerator.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text.RegularExpressions;
+
+using Microsoft.Agents.AI.Hosting.OpenAI.Responses.Models;
+
+namespace Microsoft.Agents.AI.Hosting.OpenAI.Responses.Common.Id;
+
+/// <summary>
+/// Generates Foundry-style IDs with partition keys.
+/// </summary>
+internal sealed partial class DefaultIdGenerator : IIdGenerator
+{
+    private readonly string _partitionId;
+
+#if NET9_0_OR_GREATER
+    [GeneratedRegex("^[A-Za-z0-9]*$")]
+    private static partial Regex WatermarkRegex();
+#else
+    private static readonly Regex s_watermarkRegex = new("^[A-Za-z0-9]*$", RegexOptions.Compiled);
```

Comment:
Should this be + rather than * ? It may not matter, as it looks like we only use it on inputs already validated to be non-empty.